### DISCUSSION TITLE
Apple Store not sending subscription changes to server

### DISCUSSION THREAD

**Discussion by RocketFever22** [imdhemy/laravel-in-app-purchases#114] [Q&A]
Hello everyone. Great package, I could make Google Play work easy and smooth. Thanks a lot! 

But I have a problem with Apple: 

I can verify receipts, I get the correct status with all the verification fields, but I don't recieve subscription change events in my server.

- I've set the correct url in apple's site configuration (with and withou /v2 in the end of the url)
- I've set the code in `appstore_password` (is a string, right? not a .p8 file)
- I've created listeners for every apple event in the configuration file the same way I did with google, but with apple's events.

But after purchasing or cancel a suscription, no apple events are triggered or the server does not recieve them. What am I doing wrong?

Do I need some kind of JWT that I'm missing? Maybe apple events are not triggered imediately? Is there something else I need to check?

Thanks in advance, and great work!

**Comment by RocketFever22** [comment]
OK, now we found in the server that apple is working, but the server is doing a 302 redirect... only for apple route, as google works. Any idea please?

**Comment by RocketFever22** [comment]
Ok, it turns out that apple doesn't send Accept: application/json header, required by our system (Google seems to send it). This solved the issue (3 days lost...)

Thanks anyway!

**Comment by RocketFever22** [comment]
Sorry, I could not solve it. Apple is calling to the server but it is being redirected to login (302). Isn't there a flag or something in apple's side to set a header? I can't find it...

**Comment by RocketFever22** [comment]
Ok, laravel's endpoint for apple purchases is expecting : unified_receipt, unified_receipt.latest_receipt, unified_receipt.latest_receipt_info and notification_type, but apple is sending a signedPayload jwt token. 

What am I missing?

**Comment by imdhemy** [comment]
Hi @RocketFever22 
Thank you for your patience. Looks like you are using V2 notifications which are not supported by this package. You can use V1 Notifications and try again.
